modules/videoio/test/test_gstreamer.cpp

```diff
@@ -0,0 +1,259 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+#include "opencv2/videoio/videoio_c.h"
+#ifdef HAVE_GSTREAMER
+
+namespace opencv_test
+{
+typedef cv::Point_<double> Point2d;
+typedef tuple<string, string> Param;
+
+class Videoio_Gstreamer_Test : public testing::TestWithParam<Param>
+{
+protected:
+    Videoio_Gstreamer_Test()
+    {
+        format = get<0>(GetParam());
+        pattern = get<1>(GetParam());
+        size = Size(640, 480);
+        size_nv12 = Size(640, 720);
+        count_frames = 10;
+        pipeline << "videotestsrc pattern=" << pattern;
+        pipeline << " num-buffers=" << count_frames;
+        if (format == "video/x-bayer")
+        {
+            pipeline << " ! " << format;
+        }
+        else if (format == "image/jpeg")
+        {
+            pipeline << " ! jpegenc ! " << format;
+        }
+        else
+        {
+            pipeline << " ! video/x-raw, format=" << format;
+        }
+        pipeline << ", width=" << size.width << ", height=" << size.height;
+        pipeline << " ! appsink sync=false";
```

Comment:
In my experiments `sync=false` parameter omits many frames which makes the test unstable.